File: Sources/Data/models/Account.swift
import Foundation
import FirebaseFirestoreSwift

enum AccountType: String, Codable {
    case current = "CURRENT"
    case savings = "SAVINGS"
    case creditCard = "CREDIT_CARD"
    case investment = "INVESTMENT"
}

struct Account: Codable, Identifiable {
    @DocumentID var id: String? = nil
    var name: String = ""
    var initialBalance: Double = 0
    var bankId: String? = nil
    var type: AccountType = .current
    var color: String? = nil
    var includeInDash: Bool = true
    var createdAt: Date = Date()
    var updatedAt: Date = Date()

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case initialBalance = "initialBalance"
        case bankId = "bankId"
        case type
        case color
        case includeInDash = "includeInDash"
        case createdAt = "createdAt"
        case updatedAt = "updatedAt"
    }
}

File: app/src/main/kotlin/br/com/victorwads/goldenunicorn/data/models/Account.kt
package br.com.victorwads.goldenunicorn.data.models

import com.google.firebase.firestore.DocumentId
import com.google.firebase.firestore.PropertyName
import java.util.Date

data class Account(
    @DocumentId val id: String,
    @PropertyName("name") val name: String,
    @PropertyName("type") val type: Type = Type.CURRENT,
    @PropertyName("color") val color: String?,
    @PropertyName("bankId") val bankId: String?,

    @PropertyName("initialBalance") val initialBalance: Double = 0.0,
    @PropertyName("includeInDash") val includeInTotal: Boolean = false,

    @PropertyName("createdAt") val createdAt: Date = Date(),
    @PropertyName("updatedAt") val updatedAt: Date = Date()
) {
    enum class Type {
        CURRENT, SAVINGS, CREDIT_CARD, INVESTMENT
    }
}